### PR TITLE
TEST-#5340: Use dev requirements in test-ray-master to get fastparquet

### LINKED ISSUE #5340: test-ray-master failing in CI on push to master because of missing fastparquet dependency
See e.g. here: https://github.com/modin-project/modin/actions/runs/3609028986/jobs/6081974800

The reason is that `modin/pandas/test/test_io.py::TestParquet::test_read_parquet_2462[fastparquet]` requires fastparquet, but test-ray-master [uses](https://github.com/modin-project/modin/blob/b8471075248307e48bbe2b862c7fcef78af4b5f1/.github/workflows/push-to-master.yml#L20) `requirements-no-engine`, which, unlike the dev requirements like requirements-dev.txt, does not require fastparquet.

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

TEST-#5340: Use dev requirements in test-ray-master to get fastparquet

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5340 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by pyrito — COMMENTED** [review]
How were we getting away with this before?

**Comment by mvashishtha** [comment]
> How were we getting away with this before?

@pyrito environment-dev.yml is quite close to requirements-no-engine.yml. Extra dependences that I notice are `scikit-learn`, pymssql`, and `psycopg2`, `fastparquet`, `pygit2`, and `pandas-stubs`. It doesn't look like any test uses `skit-learn`. `pygit2` is needed for the release notes tool only. `pandas-stubs` I'm not sure about. `psycopg2` and `pymssql` are needed for SQL I/O tests that we run on test-ubuntu but on push-to-master.

I recall that the `fastparquet` test was added fairly recently.

**Comment by pyrito** [comment]
> > How were we getting away with this before?
> 
> @pyrito environment-dev.yml is quite close to requirements-no-engine.yml. Extra dependences that I notice are `scikit-learn`, pymssql`, and `psycopg2`, `fastparquet`, `pygit2`, and `pandas-stubs`. It doesn't look like any test uses `skit-learn`. `pygit2`is needed for the release notes tool only.`pandas-stubs`I'm not sure about.`psycopg2`and`pymssql` are needed for SQL I/O tests that we run on test-ubuntu but on push-to-master.
> 
> I recall that the `fastparquet` test was added fairly recently.

I see. We also probably turned a blind eye to the push-to-master tests 😛

**Review by pyrito — DISMISSED** [review]

**Comment by pyrito** [comment]
@mvashishtha have you checked that this change actually works?

**Review by vnlitvinov — DISMISSED** [review]
LGTM hoping it would work!

File: .github/workflows/push-to-master.yml [hunk 1]
```diff
@@ -17,14 +17,15 @@ jobs:
       - uses: conda-incubator/setup-miniconda@v2
         with:
           activate-environment: modin
-          environment-file: requirements/requirements-no-engine.yml
```
Comment (anmyachev) [inline .github/workflows/push-to-master.yml:20]: Looks like `requirements/requirements-no-engine.yml` is no longer used in Modin. Should we remove it?

**Comment by mvashishtha** [comment]
> @mvashishtha have you checked that this change actually works?

@pyrito no, but I think it should give the fastparquet dependency, so I'm pretty hopeful. Also I've tried to copy the steps from test-ubuntu, which passes in CI: https://github.com/modin-project/modin/blob/261e29606d3ff18b71396c3e2abc640f0634c5eb/.github/workflows/ci.yml#L623-L648

Unfortunately I couldn't test this locally because `conda env create -f environment-dev.yml` was giving 

```
conda.exceptions.ResolvePackageNotFound:
  - ray-default[version='>=1.4.0']
```

Since the test is already broken and it's a bit of extra work to make a workflow that does what I'm trying here in ci.yml, test that it passes, then revert the change, I'd rather just give this a try as is.

**Review by anmyachev — APPROVED** [review]
LGTM!

**Review by vnlitvinov — APPROVED** [review]

**Pull request merged**